weejson-jackson/src/main/scala/com/rallyhealth/weejson/v1/jackson/VisitorJsonGenerator.scala

```diff
@@ -28,7 +28,7 @@ class VisitorJsonGenerator[J](
 
   private val cs = new TextBufferCharSequence(Array.emptyCharArray, 0, 0)
 
-  private val stack: mutable.ArrayStack[ObjArrVisitor[Any, _]] = new mutable.ArrayStack[ObjArrVisitor[Any, _]]()
+  private var stack: List[ObjArrVisitor[Any, _]] = Nil
```

Comment:
@htmldoug, so good to hear from you (I did not expect it)!!!

I was pretty much blindly following the direction of the compiler(s) here - if you think the change will be for the worse, I will revert it and ignore the deprecation. Scala 2.13 (and I assume Scala 3 as well) deprecated `ArrayStack` with the message "Use `Stack` instead of `ArrayStack`; it now uses an array-based implementation". But if I do that, then Scala 2.12 complains that "`Stack` in package mutable is deprecated (since 2.12.0): `Stack` is an inelegant and potentially poorly-performing wrapper around `List`. Use a `List` assigned to a var instead." So that's what I did.

Reply:
Good to hear from you, too!

If you want to be diligent, you could measure before/after. Since scala reimplemented `Stack` to use `Array` instead of `List`, maybe performance was one of the reasons.

IIRC, the difference in throughput was negligible in benchmarks, although maybe there are different conditions where it matters. I kinda went after all allocation indiscriminately since it always eliminated GC pauses, sometimes improved throughput, and most of all, removed allocation noise from the profiler that I no longer had to scrutinize further as "okay" or "bad".

Worst case scenario, it's `private`, so it can be changed later without breaking binary compatibility.

I'd say go ahead using the `List`. Admittedly, I have much less skin in the game now that I'm no longer a service owner.